### PR TITLE
Grab package gitsha even in package subdirectories

### PR DESCRIPTION
**Pull request by staticfloat**
This is a slightly more gung-ho method of attempting to find the package gitsha.

### CODE REVIEW

File: src/02_environment.jl [hunk 1]
```diff
@@ -42,18 +42,10 @@ immutable Environment
...
-            if isempty(sha1)
-                Nullable{UTF8String}()
-            else
-                Nullable{UTF8String}(utf8(sha1))
-            end
-        else
-            Nullable{UTF8String}()
+        package_sha1 = Nullable{UTF8String}()
+        try
+            sha1 = readall(pipeline(`git rev-parse HEAD`, stderr=Base.DevNull))
```
Comment (johnmyleswhite) [inline src/02_environment.jl]: Seems like this needs a chomp

File: src/02_environment.jl [hunk 2]
```diff
@@ -42,18 +42,10 @@ immutable Environment
         uuid = string(Base.Random.uuid4())
         timestamp = Libc.strftime("%Y-%m-%d %H:%M:%S", round(Int, time()))
         julia_sha1 = Base.GIT_VERSION_INFO.commit
-        package_sha1 = if isdir(".git")
-            sha1 = ""
-            try
-                sha1 = readchomp(`git rev-parse HEAD`)
-            end
-            if isempty(sha1)
```
Comment (johnmyleswhite) [inline src/02_environment.jl:50]: What happens in this case now? You get a non-null result, but it's empty?

File: src/02_environment.jl (hunk 1 above)
Comment (staticfloat) [inline src/02_environment.jl]: Fair enough; I didn't have excess whitespace when I tested, but it can't hurt.

File: src/02_environment.jl (hunk 2 above)
Comment (staticfloat) [inline src/02_environment.jl:50]: I don't think we can have an empty result; if `git rev-parse HEAD` doesn't spit something out, it throws an exception, so we continue along with the original value of `package_sha1`, which is `NULL`.

**Comment by johnmyleswhite** [comment]
Ok, looks good to go to me. Squash and merge?

**Comment by staticfloat** [comment]
Done.

**Pull request merged**